src/util.cpp

```diff
@@ -662,26 +661,46 @@ fs::path GetConfigFile(const std::string& confPath)
     return AbsPathForConfigVal(fs::path(confPath), false);
 }
 
+inline bool not_space(int c) { return !std::isspace(c); }
+
 void ArgsManager::ReadConfigFile(const std::string& confPath)
 {
-    fs::ifstream streamConfig(GetConfigFile(confPath));
-    if (!streamConfig.good())
+    fs::ifstream config_file(GetConfigFile(confPath));
+    if (!config_file.good())
         return; // No bitcoin.conf file is OK
 
     {
+        // left and right trim for strings
+        auto ltrim = [](std::string &s) { s.erase(s.begin(), std::find_if(s.begin(), s.end(), not_space)); };
+        auto rtrim = [](std::string &s) { s.erase(std::find_if(s.rbegin(), s.rend(), not_space).base(), s.end()); };
+
         LOCK(cs_args);
-        std::set<std::string> setOptions;
-        setOptions.insert("*");
+        std::string line;
+        while (std::getline(config_file, line)) {
+            size_t eqpos = line.find('=');
```

Comment:
Current syntax seems to be described here:

http://www.boost.org/doc/libs/1_66_0/doc/html/program_options/overview.html#id-1.3.31.5.10.2

I wonder if for backwards compatibility or convenience the new implementation should be extended to support `#` comments.

Reply:
+1